### PR TITLE
bugfix: CLDSRV-11 Write ACL via predefined groups

### PR DESCRIPTION
**Pull request by miniscruff**

### CODE REVIEW

**Comment by miniscruff** [comment]
ping

**Comment by miniscruff** [comment]
reset

**Review by naren-rajendran — APPROVED** [review]

File: lib/api/apiUtils/authorization/aclChecks.js [hunk 1]
```diff
@@ -73,6 +74,13 @@ function isBucketAuthorized(bucket, requestType, canonicalID) {
             || bucketAcl.FULL_CONTROL.indexOf(canonicalID) > -1
             || bucketAcl.WRITE.indexOf(canonicalID) > -1) {
             return true;
+        } else if (bucketAcl.WRITE.indexOf(publicId) > -1
```
Comment (alexanderchan-scality) [inline lib/api/apiUtils/authorization/aclChecks.js:77]: both of the blocks return `true`, so they can be merged.

File: lib/api/objectDelete.js [hunk 2]
```diff
@@ -55,6 +55,12 @@ function objectDelete(authInfo, request, log, cb) {
                 if (err) {
                     return next(err, bucketMD);
                 }
+                // based on AWS's behavior, object delete can be performed by
+                // either bucket owner or the object owner
+                if ((objMD && objMD['owner-id'] !== canonicalID)
+                    && bucketMD.getOwner() !== canonicalID) {
```
Comment (alexanderchan-scality) [inline lib/api/objectDelete.js:61]: ~would this conditional affect the `public-read-write` behavior?~

File: lib/api/apiUtils/authorization/aclChecks.js (hunk 1 above)
Comment (miniscruff) [reply lib/api/apiUtils/authorization/aclChecks.js:77]: I believe this was to match the other conditionals in the method, I cherry-picked it from a commit by @naren-scality

File: lib/api/objectDelete.js (hunk 2 above)
Comment (naren-rajendran) [reply lib/api/objectDelete.js:61]: @alexanderchan-scality how do you think this would affect?

Comment (alexanderchan-scality) [reply lib/api/objectDelete.js:61]: The comment is not valid. I was under the impression that the `public-read-write' should allow other users to overwrite object even when they don't own it. Which is not the case in real AWS as mentioned in the PR. double-checked against real AWS
the check is still valid.

Comment (alexanderchan-scality) [reply lib/api/objectDelete.js:61]: this check can however added as part of the `isObjAuthorized` method as opposed to each of the api handlers

Comment (naren-rajendran) [reply lib/api/objectDelete.js:61]: This was very specific to object delete and wanted to avoid conditionals in `isObjAuthorized`.

Comment (alexanderchan-scality) [reply lib/api/objectDelete.js:61]: i see. does the check only occur for the `objectPut` and `objectDelete` operation? I was wondering about the other api calls that would result in object overwrites like `objectCopy`

Comment (naren-rajendran) [reply lib/api/objectDelete.js:61]: `objectCopy` is a combo of `objectGet` and `objectPut` so the user performing the action should have permission to read and write and `complete-mpu` is same as objectPutPart which again internally is an `objectPut`

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**